app/interactors/album.js

```diff
@@ -0,0 +1,25 @@
+const Album = require('../models').albums;
+
+exports.purchaseAlbum = (user, album) => {
+
+  return new Promise((resolve, reject) => {
+
+    Album.findOne({ where: {id: album.id, userId : user.id} }).then(alreadyPurchased => {
+
+      if(alreadyPurchased){
+        resolve(undefined);
```

Comment:
I don't like this very much...
I think you have 3 options here:
- change this undefined for an empty object {} and leave the rest the same.
- Make this function to return a boolean indicating the actual result of the purchase.
- Make this function throw and error in case you have already purchased that book.